.travis.yml

```diff
@@ -1,19 +1,15 @@
-# see http://about.travis-ci.org/docs/user/languages/php/ for more hints
 language: php
 
-# list any PHP version you want to test against
 php:
-  # using major version aliases
-
-  # aliased to 5.3.29
+  - 5.3.3
   - 5.3
-  # aliased to a recent 5.4.x version
   - 5.4
-  # aliased to a recent 5.5.x version
   - 5.5
-  # aliased to a recent 5.6.x version
   - 5.6
+  - hhvm
   
-before_script: composer install  
+sudo: false
+
+install: travis_retry composer install --no-interaction --prefer-source
```

Comment:
Regarding the third comment, there is a better way given that the new travis infrastructure supports caching: let composer use dists for stable versions but put the ``$HOME/.composer/cache`` folder in the build cache, so that most builds won't need the API. It makes things faster.